src/realm/group_shared.cpp

```diff
@@ -411,7 +412,7 @@ struct alignas(8) SharedGroup::SharedInfo {
     ///
     /// CAUTION: This member must never move or change type, as that would
     /// compromize safety of the the session initiation process.
-    uint8_t init_complete = 0; // Offset 0
+    std::atomic<bool> init_complete; // Offset 0
```

Comment:
I think we need to leave it explicitly `uint8_t`. The standard 5.3.3/1 says "The result of sizeof applied to any other fundamental type (3.9.1) is implementation-defined. [ Note: in particular, sizeof(bool), sizeof(char16_t), sizeof(char32_t), and sizeof(wchar_t) are implementation-defined.74 —end note ] ... Note: 74) sizeof(bool) is not required to be 1." If some implementation has a different size for bools, this change would not be backwards compatible.

Reply:
Also note that from section 29.5/9 there is no guarantee that `sizeof(std::atomic<T>) == sizeof(T)`.
"[ Note: The representation of an atomic specialization need not have the same size as its corresponding argument type. Specializations should have the same size whenever possible, as this reduces the effort required to port existing code. —end note ]"

Reply:
Wrt to sizeof(std::atomic<T>)==sizeof(T) .. we know there is no guarantee, which is why we have static asserts which will fail on platforms, where the sizes doesn't match. You may be right that using atomic<uint8_t> is better.